Kernel/Net/NetworkAdapter.cpp

```diff
@@ -169,4 +169,34 @@ void NetworkAdapter::set_ipv6_netmask(IPv6Address const& netmask)
     m_ipv6_netmask = netmask;
 }
 
+void NetworkAdapter::autoconfigure_link_local_ipv6()
+{
+    auto mac = mac_address();
+    if (mac.is_zero() || !link_up())
+        return;
+
+    // TODO: other IPv6 autoconf modes
+    // TODO: duplicate address detection, this is only a very naive implementation of autoconf
+    auto ipv6_ll = IPv6Address({ 0xfe,
+        0x80,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        static_cast<u8>(mac[0] ^ 0b00000100),
```

Comment:
why the xor?

Reply:
This is mandated by the conversion from an IEEE MAC-48 identifier to an EUI-64 identifier (Appendix A of RFC4291).

There actually is an error in there too: the 7th bit and not the 6th bit should be inverted.

Reply:
(could've added a spec comment for this. Maybe in the next one.)